src/PlexShareScreenshare/Server/SharedClientScreen.cs

```diff
@@ -0,0 +1,357 @@
+                }
+            }
+        }
+
+        /// <summary>
+        /// Insert the received Frame into the received image queue.
+        /// </summary>
+        /// <param name="frame">
+        /// Frame to be inserted.
+        /// </param>
+        public void PutImage(Frame frame)
+        {
+            Debug.Assert(_imageQueue != null, GetDebugMessage("_imageQueue is found null"));
+
+            lock (_imageQueue)
+            {
+                _imageQueue.Enqueue(frame);
+            }
+        }
+
+        /// <summary>
+        /// Pops and returns the final Image at the beginning of the final image queue.
+        /// </summary>
+        /// <returns>
+        /// The final image to be displayed that is removed from the beginning.
+        /// </returns>
+        public Bitmap? GetFinalImage()
+        {
+            Debug.Assert(_finalImageQueue != null, GetDebugMessage("_finalImageQueue is found null"));
+
+            // Wait until the queue is not empty
+            while (_finalImageQueue.Count != 0) Thread.Sleep(100);
+
+            lock (_finalImageQueue)
+            {
+                try
+                {
+                    return _finalImageQueue.Dequeue();
+                }
+                catch (InvalidOperationException e)
+                {
+                    Trace.WriteLine(GetDebugMessage($"Dequeue failed: {e.Message}", withTimeStamp: true));
+                    return null;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Insert the final image into the final image queue.
+        /// </summary>
+        /// <param name="image">
+        /// Image to be inserted.
+        /// </param>
+        public void PutFinalImage(Bitmap image)
+        {
+            Debug.Assert(_finalImageQueue != null, GetDebugMessage("_finalImageQueue is found null"));
+
+            lock (_finalImageQueue)
+            {
+                _finalImageQueue.Enqueue(image);
+            }
+        }
+
+        /// <summary>
+        /// Starts image processing by calling the underlying stitcher to process images.
+        /// It will also create (if not exist) and start the task for updating the displayed
+        /// images and notify the UX.
+        /// </summary>
+        /// <param name="task"></param>
+        /// <exception cref="OperationCanceledException"></exception>
+        public void StartProcessing(Action<CancellationToken> task)
+        {
+            Debug.Assert(_stitcher != null, GetDebugMessage("_stitcher is found null"));
+
+            // Start the stitcher
+            _stitcher.StartStitching();
+
+            if (_imageSendTask == null)
+            {
+                // Create a new cancellation token
+                _source = new CancellationTokenSource();
+                try
+                {
+                    // Create and start a new task
+                    _imageSendTask = new Task(() => task(_source.Token), _source.Token);
```

Comment:
I don't get how this works. Where's the lambda/function that is to be passed in the task?

Reply:
It will be passed by the view model.